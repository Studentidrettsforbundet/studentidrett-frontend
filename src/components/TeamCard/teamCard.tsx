import React from 'react';
import image from '../../assets/placeholder.png';
import { card, cardBody, imgContainer, cardImg, cardTitle } from '../../styles/card';
import { resetFetchStatusesActionCreator } from '../../store/thunks/thunkActions';
import { Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';

import { InfoList } from '../TeamInfo/infoList';
import { teamInterface } from '../../interfaces';

const TeamCard = ({ id, name, short_description, gender, availability, skill_level }: teamInterface): JSX.Element => {
    const dispatch = useDispatch();

    return (
        <Link
            to={`/teams/${id}`}
            key={id}
            className={'unstyled_link'}
            onClick={() => {
                dispatch(resetFetchStatusesActionCreator());
            }}
        >
            <div className={card} key={id}>
                <div className={imgContainer}>
                    <img className={cardImg} src={image} alt="Team"></img>
                </div>
                <div className={cardBody}>
                    <h5 className={cardTitle}>{name}</h5>
                    <InfoList availability={availability} gender={gender} skill_level={skill_level} />
                    <p>{short_description}</p>
                </div>
            </div>
        </Link>
    );
};

export default TeamCard;
