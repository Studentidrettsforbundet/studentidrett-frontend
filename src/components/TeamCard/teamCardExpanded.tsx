import React from 'react';
import { getLinkIcon } from '../../services/iconCreator';

export interface teamCardInterface {
    long_description?: string;
    gender?: string;
    skill_level?: string;
    availability?: string;
    facebook_link?: string;
    instagram_link?: string;
    webpage?: string;
}

const TeamCardExpanded = ({
    long_description,
    gender,
    skill_level,
    availability,
    facebook_link,
    instagram_link,
    webpage,
}: teamCardInterface): JSX.Element => {
    const a = 'http://www.facebook.com/';
    const renderLinks = () => {
        const links = [
            { site: a, text: 'Facebook' },
            { site: a, text: 'Instagram' },
            { site: a, text: 'Hjemmeside' },
        ];
        if (a || instagram_link || webpage) {
            return (
                <div className="list">
                    {links.map((link, i) => {
                        if (link.site) {
                            return (
                                <a className="primary link-item" key={i} href={link.site}>
                                    <div className="icon">{getLinkIcon(link.text)}</div>
                                    <div className="link-text">{link.text}</div>
                                </a>
                            );
                        }
                    })}
                </div>
            );
        }
    };
    return (
        <div>
            <p>{long_description}</p>
            {renderLinks()}
        </div>
    );
};

export default TeamCardExpanded;
