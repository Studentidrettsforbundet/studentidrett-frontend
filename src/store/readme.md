# Redux

This folder contains all redux files.

Add this extension to your browser: https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd

## Description:

store.ts is the file that creates the redux store, as well as the combined reducer. All reducers needs to be added to the combined reducer, and to the combinedState

This folder contains a folder for each page of our application that needs redux.
The folders contains reducers, actions, action crators, types, interfaces, etc. related to pages.

## Example of folder structure

We have a page called "search". That page needs a reducer, some actions, action creators, and some interfaces. Below is an example of the folder structure

----/store
--------store.ts
--------/pages
-----------/search
-----------------searchReducer.ts
-----------------searchActions.ts
-----------------searchReducer.ts
