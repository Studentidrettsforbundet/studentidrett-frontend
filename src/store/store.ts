import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import { cityInitialState, cityReducer, cityState } from './pages/city/cityReducer';
import { clubInitialState, clubReducer, clubDetailReducer, clubState } from './pages/club/clubReducer';
import { groupInitialState, groupReducer, groupDetailReducer, groupState } from './pages/group/groupReducer';
import { regionInitialState, regionReducer, regionState } from './pages/region/regionReducer';
import { sportInitialState, sportReducer, sportDetailReducer, sportState } from './pages/sport/sportReducer';
import { teamInitialState, teamReducer, teamDetailReducer, teamState } from './pages/team/teamReducer';
import { searchBarInitialState, searchBarReducer, searchBarState } from './searchBar/searchBarReducer';
import { thunkInitialState, thunkReducer, thunkState } from './thunks/thunkReducer';
import { searchInitialState, searchReducer, searchState } from './pages/search/searchReducer';
import {
    questionInitialState,
    questionnaireReducer,
    questionnaireState,
} from './pages/questionnaire/questionnaireReducer';

declare global {
    interface Window {
        __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
    }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

//TODO: add reducers interfaces
export interface combinedStateInterface {
    region: regionState;
    city: cityState;
    sport: sportState;
    sport_detail: sportState;
    club: clubState;
    club_detail: clubState;
    team: teamState;
    team_detail: teamState;
    group: groupState;
    group_detail: groupState;
    search_results: searchState;
    thunk: thunkState;
    searchBar: searchBarState;
    questionnaire: questionnaireState;
}

// TODO: add all pages state here
export const combinedState = {
    region: regionInitialState,
    city: cityInitialState,
    sport: sportInitialState,
    sport_detail: sportInitialState,
    club: clubInitialState,
    club_detail: clubInitialState,
    team: teamInitialState,
    team_detail: teamInitialState,
    group: groupInitialState,
    group_detail: groupInitialState,
    search_results: searchInitialState,
    thunk: thunkInitialState,
    searchBar: searchBarInitialState,
    questionnaire: questionInitialState,
};

// create store
const store = createStore(
    combineReducers({
        region: regionReducer,
        city: cityReducer,
        sport: sportReducer,
        sport_detail: sportDetailReducer,
        club: clubReducer,
        club_detail: clubDetailReducer,
        team: teamReducer,
        team_detail: teamDetailReducer,
        group: groupReducer,
        group_detail: groupDetailReducer,
        search_results: searchReducer,
        thunk: thunkReducer,
        searchBar: searchBarReducer,
        questionnaire: questionnaireReducer,
    }),
    combinedState,
    composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
