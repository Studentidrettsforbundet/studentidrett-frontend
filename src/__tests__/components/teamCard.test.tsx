import React, { useState } from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import TeamCard from '../../components/TeamCard/teamCard';
import store from '../../store/store';

describe('teamCard', () => {
    test('renders correctly', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/teams/1']}>
                    <TeamCard
                        id={0}
                        name={'Men A'}
                        gender={'MX'}
                        short_description={'This is where the short description goes'}
                        availability={'OP'}
                        skill_level={'PRO'}
                    />
                </MemoryRouter>
            </Provider>,
        );
        expect(tree).toMatchSnapshot();
    });
});
