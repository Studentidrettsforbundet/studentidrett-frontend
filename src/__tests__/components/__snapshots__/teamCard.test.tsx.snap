// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teamCard renders correctly 1`] = `
<a
  className="unstyled_link"
  href="/teams/0"
  onClick={[Function]}
>
  <div
    className="f1bqh63k"
  >
    <div
      className="f1gj5ds8"
    >
      <img
        alt="Team"
        className="fb12b09"
        src="placeholder.png"
      />
    </div>
    <div
      className="f9hjpet"
    >
      <h5
        className="f1rpkic8"
      >
        Men A
      </h5>
      <div
        className="fppg92i"
      >
        <div
          className="f1ubk0hm"
        >
          <div>
            <b>
              Status
              :
            </b>
             
            Åpent
          </div>
          <div
            className="f1ozov1j"
          >
            <svg
              className="bi bi-person-plus-fill"
              fill="currentColor"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H1zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6zm7.5-3a.5.5 0 0 1 .5.5V7h1.5a.5.5 0 0 1 0 1H14v1.5a.5.5 0 0 1-1 0V8h-1.5a.5.5 0 0 1 0-1H13V5.5a.5.5 0 0 1 .5-.5z"
                fillRule="evenodd"
              />
            </svg>
          </div>
        </div>
        <div
          className="f1ubk0hm"
        >
          <div>
            <b>
              Kjønn
              :
            </b>
             
            Mix
          </div>
          <div
            className="f1ozov1j"
          >
            <svg
              fill="currentColor"
              stroke="currentColor"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Male Female
              </title>
              <path
                d="M426 16h-74a22 22 0 000 44h20.89l-37.1 37.09A157.68 157.68 0 00216 42c-87.12 0-158 70.88-158 158 0 79.66 59.26 145.72 136 156.46V394h-28a22 22 0 000 44h28v36a22 22 0 0044 0v-36h28a22 22 0 000-44h-28v-37.54c76.74-10.74 136-76.8 136-156.46a157.15 157.15 0 00-14-64.92l44-44V112a22 22 0 0044 0V38a22 22 0 00-22-22zM216 314a114 114 0 11114-114 114.13 114.13 0 01-114 114z"
              />
            </svg>
          </div>
        </div>
        <hr />
      </div>
      <p>
        This is where the short description goes
      </p>
    </div>
  </div>
</a>
`;
